### PR TITLE
Sales analyst

### PR DESCRIPTION
**Pull request by rcasias**
Created all methods for sales analyst in iteration 1. Our test is at about 1 min run time, maybe look at ways to reduce this in future.

### CODE REVIEW

File: lib/sales_analyst.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  def average_items_per_merchant_standard_deviation
+    squared_differences = items_per_merchant.map do |num|
+      ((num - average_items_per_merchant)**2).to_f
+    end.sum
+    divided_sum = ((squared_differences) / (merchant_id_array.length - 1))
+    square_root = ((divided_sum)**0.5).to_f
+    square_root.round(2)
+  end
+
+  def z_score_items(value)
```
Comment (deebot10) [inline lib/sales_analyst.rb:50]: z_score working fine this method is perfect for making calculations with standard deviation

File: lib/sales_analyst.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    merchant_id_array.map do |id|
+      @items_repo.find_all_by_merchant_id(id).length
+    end
+  end
+
+  def average_items_per_merchant_standard_deviation
+    squared_differences = items_per_merchant.map do |num|
+      ((num - average_items_per_merchant)**2).to_f
+    end.sum
+    divided_sum = ((squared_differences) / (merchant_id_array.length - 1))
```
[the remark this answers is not in the document]
Comment (ahenzler) [reply lib/sales_analyst.rb:45]: maybe remove extra parenthesses here

File: spec/sales_analyst_spec.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    end
+
+    it 'can find the average price of a merchants items' do
+
+      expect(sales_analyst.average_item_price_for_merchant(12334105).class).to eq(BigDecimal)
+      expect(sales_analyst.average_item_price_for_merchant(12334105)).to eq(0.1666e2)
+    end
+
+    it 'can find the average of average price' do
+      expect(sales_analyst.average_average_price_per_merchant.class).to eq(BigDecimal)
```
[the remark this answers is not in the document]
Comment (ahenzler) [reply spec/sales_analyst_spec.rb:55]: lets rename long method names  when we refactor

**Review by ahenzler — APPROVED** [review]
When we refactor 
- look for extra parentheses 
- extra whitespace 
- commented out stuff
- rename methods with long names

File: lib/sales_analyst.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    square_root = ((divided_sum)**0.5).to_f
+    square_root.round(2)
+  end
+  def z_score(value)
+    ((value - average_items_per_merchant) / average_items_per_merchant_standard_deviation).to_f
```
[the remark this answers is not in the document]
Comment (rcasias) [reply lib/sales_analyst.rb:51]: we can remove the extra set of parenthesis here.

**Review by deebot10 — APPROVED** [review]

File: lib/sales_analyst.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      @items_repo.find_all_by_merchant_id(id).length
+    end
+  end
+
+  def average_items_per_merchant_standard_deviation
+    squared_differences = items_per_merchant.map do |num|
+      ((num - average_items_per_merchant)**2).to_f
+    end.sum
+    divided_sum = ((squared_differences) / (merchant_id_array.length - 1))
+    square_root = ((divided_sum)**0.5).to_f
```
[the remark this answers is not in the document]
Comment (rcasias) [reply lib/sales_analyst.rb:46]: we can remove the extra set of parentheses  here

**Comment by rcasias** [comment]
We should probably revisit and refactor some of these methods.

**Pull request merged**